### PR TITLE
feat(passengers): adds ability to add passengers to rides

### PR DESCRIPTION
**Pull request by codyleyhan**
# Description
- added passengers to rides
- adds a PR template
screen shot 2018-04-27 at 11 16 29 pm

# Steps to Reproduce
- hit the resource at /api/v1/passengers

# Tested
- fully unit tested
screen shot 2018-04-27 at 11 15 08 pm

# Approved
- [x] @jcherianucla 
- [x] @epicchewy

### CODE REVIEW

File: services/passenger.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+}
+
+// GetAll returns a page of rides
+func (p *PassengerService) GetAll(lastID string, limit, userAuthLevel int) ([]*models.Passenger, error) {
+	if userAuthLevel < AdminLevel {
+		return nil, ErrNotAllowed
+	}
+
+	if limit <= 0 || limit > 100 {
+		limit = 15
```
Comment (jcherianucla) [inline services/passenger.go]: Make this a constant

File: services/passenger.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+	return p.store.GetAll(lastID, limit)
+}
+
+// GetAllByCarID returns all passengers in all statuses for a given car
+func (p *PassengerService) GetAllByCarID(carID string) ([]*models.Passenger, error) {
+	return p.store.WhereMany([]stores.QueryModifier{stores.QueryMod("car_id", stores.EQ, carID)})
+}
+
+// Delete removes a ride from the store if the user is allowed to
```
Comment (jcherianucla) [inline services/passenger.go]: nit: Do you mean it removes a passenger?

**Review by jcherianucla — COMMENTED** [review]
Overall looks good to me

**Comment by epicchewy** [comment]
lgtm

**Review by epicchewy — APPROVED** [review]

File: stores/postgres/passenger_sql.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	FOREIGN KEY (ride_id) REFERENCES rides (id) ON DELETE CASCADE,
+	UNIQUE (driver_id, passenger_id, ride_id)
+);`
+
+	passengerGetAllSQL = "SELECT * FROM passengers WHERE id > $1 LIMIT $2"
+
+	passengerGetByIDSQL = "SELECT * FROM passengers WHERE id=$1"
+
+	passengerInsertSQL = "INSERT INTO passengers (id, driver_id, passenger_id, ride_id, status) VALUES ($1, $2, $3, $4, $5) RETURNING created_at, updated_at"
+	passengerUpdateSQL = "UPDATE passengers SET status=$1 updated_at=NOW() WHERE id=$12 RETURNING updated_at"
```
Comment (epicchewy) [inline stores/postgres/passenger_sql.go]: should be WHERE id=$2

**Review by epicchewy — DISMISSED** [review]
just found something in the sql

**Pull request merged**